src/strgen/strgen_base.cpp

```diff
@@ -38,50 +38,29 @@ static const CmdStruct *ParseCommandString(const char **str, char *param, int *a
  * Create a new case.
  * @param caseidx The index of the case.
  * @param string  The translation of the case.
- * @param next    The next chained case.
  */
-Case::Case(int caseidx, const char *string, Case *next) :
-		caseidx(caseidx), string(stredup(string)), next(next)
+Case::Case(int caseidx, const std::string &string) :
+		caseidx(caseidx), string(string)
 {
 }
 
-/** Free everything we allocated. */
-Case::~Case()
-{
-	free(this->string);
-	delete this->next;
-}
-
 /**
  * Create a new string.
  * @param name    The name of the string.
  * @param english The english "translation" of the string.
  * @param index   The index in the string table.
  * @param line    The line this string was found on.
  */
-LangString::LangString(const char *name, const char *english, size_t index, int line) :
-		name(stredup(name)), english(stredup(english)), translated(nullptr),
-		hash_next(0), index(index), line(line), translated_case(nullptr)
-{
-}
-
-/** Free everything we allocated. */
-LangString::~LangString()
+LangString::LangString(const std::string &name, const std::string &english, size_t index, int line) :
+		name(name), english(english), index(index), line(line)
```

Comment:
Do we not normally put the `:` on the newline, rather than before?

```suggestion
LangString::LangString(const std::string &name, const std::string &english, size_t index, int line)
		: name(name), english(english), index(index), line(line)
```

Reply:
```
src$ grep -R ':$' .|grep ::|grep -v 3rdparty|grep -v case|wc -l
75
src$ grep -R '       : ' .|grep -v 3rdparty|wc -l
56
```
So... the `:` at the end of the line seems more prevalent than at the begin of the line. But there is a clear lack of consensus about it.